Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAMSIConfigurationState.ps1

```diff
@@ -0,0 +1,76 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+. $PSScriptRoot\..\..\Helpers\Invoke-CatchActions.ps1
+Function Get-ExchangeAMSIConfigurationState {
+    [CmdletBinding()]
+    param ()
+
+    begin {
+        Function Get-AMSIStatusFlag {
+            [CmdletBinding()]
+            [OutputType([bool])]
+            param (
+                [Parameter(Mandatory = $true)]
+                [object]$AMSIParameters
+            )
+
+            try {
+                Switch ($AMSIParameters.Split("=")[1]) {
+                    ("False") { return $false }
+                    ("True") { return $true }
+                    Default { return $null }
+                }
+            } catch {
+                throw
+            }
+        }
+
+        Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+        $amsiState = "Unknown"
+        $amsiOrgWideSetting = $true
+        $amsiConfigurationQuerySuccessful = $false
+    } process {
+        try {
+            Write-Verbose "Trying to query AMSI configuration state"
+            $amsiConfiguration = Get-SettingOverride -ErrorAction Stop | Where-Object { ($_.ComponentName -eq "Cafe") -and ($_.SectionName -eq "HttpRequestFiltering") }
+            $amsiConfigurationQuerySuccessful = $true
+
+            if ($null -ne $amsiConfiguration) {
+                Write-Verbose "$($amsiConfiguration.Count) override(s) detected for AMSI configuration"
+                $amsiMultiConfigObject = @()
+                foreach ($amsiConfig in $amsiConfiguration) {
+                    $amsiState = Get-AMSIStatusFlag -AMSIParameters $amsiConfig.Parameters -ErrorAction Stop
```

Comment:
Since this has a possible `throw` in it and we are calling a few times. Maybe we should do a `try` `catch` here to handle this. That way we don't fail on all the possible results here.

Reply:
Good finding. Wrapped the `Get-AmsiStatusFlag` into an `try` `catch` block.
We do the following within the `catch`:
```
Write-Verbose "Unable to process: $($amsiConfig.Parameters) to determine status flags"
$amsiState = "Unknown"
Invoke-CatchActions
```

Need to set the `$amsiState` to `Unknown` again to make sure that a value which was (possibly) set in a previous call (if the exception happens between the processing of multiple objects) is overwritten.